recipes/cuda-kat/all/conanfile.py

```diff
@@ -29,26 +33,44 @@ def _compilers_minimum_version(self):
             "Visual Studio": "15",
             "clang": "5",
             "apple-clang": "10",
-            }
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
 
     def validate(self):
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("CUDA-kat library are not compatible on Windows")
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 17)
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
         minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if not minimum_version:
-            self.output.warn(f"{self.name} requires C++17. Your compiler is unknown. Assuming it supports C++17.")
-        elif tools.Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration(f"{self.name} requires a compiler that supports at least C++17")
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def package(self):
-        self.copy(pattern="LICENSE*", src=self._source_subfolder, dst="licenses")
-        self.copy("*", src=os.path.join(self._source_subfolder, "src"), dst="include")
+        copy(
+            self,
+            pattern="LICENSE",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+        )
+        copy(
+            self,
+            pattern="*",
+            dst=os.path.join(self.package_folder, "include"),
+            src=os.path.join(self.source_folder, "src"),
+        )
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "cuda-kat"
-        self.cpp_info.names["cmake_find_package_multi"] = "cuda-kat"
+        self.cpp_info.bindirs = []
+        self.cpp_info.libdirs = []
+
+        if is_msvc(self):
+            self.cpp_info.cxxflags.append("/Zc:__cplusplus")
```

Comment:
This is the typical line that we would appreciate a bit more of context, like one comment explaining why this flag is necessary, in what cases it fails if it is not there. For this one, it is not very clear why the consumers would really care about the pre-processor macro, if they rely on it, it should be their build systems that activate it?

Reply:
Sure. I briefly mentioned the reasoning in the commit message, but this is a much better location for it, I agree. Added it here as well.

Reply:
That helps a lot to simplify future maintenance, being able to read things directly in recipe, saves a lot of hops to understand things, many thanks!